File: Python/Problems/Problems/problem2027.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem2027(Problem):
    def test(self) -> bool:
        s = "XXOX"

        expected = 2

        result = self.minimumMoves(s)

        print(result)

        return expected == result

    def minimumMoves(self, s: str) -> int:
        n = len(s)
        result = 0
        i = 0
        while i < n:
            if s[i] == 'X':
                result += 1
                i += 2
            i += 1

        return result

File: C++/Problems/Problems/problem_2027.cpp
#include "problem_2027.h"

bool problem_2027::test() {
    std::string s = "XXOX";

    int expected = 2;

    auto result = minimumMoves(s);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_2027::minimumMoves(std::string s) {
    int n = s.size();
    int result = 0;
    for (int i = 0; i < n; ++i)
    {
        if (s[i] == 'X')
        {
            result++;
            i += 2;
        }
    }
    return result;
}
